### DISCUSSION TITLE
During plugin simulation, the private properties are getting reset to default(instead of mocked values)

### DISCUSSION THREAD

**Discussion by AnoopRawat** [jordimontana82/fake-xrm-easy#599] [General]
I've private property `'_seriesNumber'` in my plugin `'SeriesSaveAction'`, I mock this property before PluginSimulation but it always reset to null during simulation pipeline, while during call to other method of entity, this property shows correct mocked value.

**Below is sample code**

Entity code : 
```
public class Series : : EntityBaseClass
{
               // some other prop, Ctor & methods.
               
		private string _seriesNumber; 
				
		public void SetSeriesNumber(string seriesNo)
		{
			_seriesNumber= seriesNo; // setting private prop
		}
						
		public bool ValidSeriesNumber()
		{
			return _seriesNumber?.Length == 9;
		}	
}
```

Unit test method is like :
```
        [TestMethod]
        public void PlugInSimulation_SeriesSaveActions()
        {

           //_service & _context are already setup in another FakeXrmEasyPipelineTestsBase class.
            var fakeProxy = new XrmServiceProxy(_service);

            var target = new Series(Guid.NewGuid());
            target.SetSeriesNumber("123456789");  // set mocked value as "123456789"

            _context.RegisterPluginStep<SeriesSaveActions>(new PluginStepDefinition()
            {
                MessageName = "Create", // tried also with Update ?
                EntityLogicalName = target.EntityLogicalName,
                Stage = ProcessingStepStage.Preoperation, // also tried with Preoperation
            });

             // now the plugin simulation hits but _seriesNumber is Null 
            // ERROR: private property reset to null during plugin simulation.
            target.CreateSelf(fakeProxy, true); 

            // SUCCESS : when the another method of entity called, then private property persist it's mocked value "123456789"
            bool result = target.ValidSeriesNumber();
            Xunit.Assert.True(result);
          }
```

In plugin simulation, at some point plugin calls `ValidSeriesNumber(),` but the private property is always Null inside this method, however it should be showing-up mocked value.

**Question:**
why the private property is null during plugin simulation, but has mocked value during normal method call ?
how to correctly mock during plugin simulation pipeline ? am I missing something ?

**Comment by jordimontana82** [comment]
@AnoopRawat Thank you for raising this. I'll move this discussion into the DynamicsValue/fake-xrm-easy repo. This discussion belongs to the deprecated version v1 repo which doesn't have this functionality.

**Comment by jordimontana82** [comment]
@AnoopRawat I can't actually transfer the discussion because it belongs to a different org, could we please continue the discussion here? https://github.com/DynamicsValue/fake-xrm-easy/discussions/67

**Comment by jordimontana82** [comment]
I'm closing this discussion now, feel free to raise any questions ion the other discussion link.
